protocol/tx/entry.go

```diff
@@ -73,6 +97,10 @@ func writeForHash(w io.Writer, c interface{}) error {
 	case bc.AssetID:
 		_, err := w.Write(v[:])
 		return errors.Wrap(err, "writing bc.AssetID for hash")
+
+	case *EntryRef:
```

Comment:
There are a couple of problems with this suggestion: first, if the hash lives in an `entry`, then you need special separate handling for the case where there's only a hash and no `entry`. Second, you may recall that we experimented with this while doing the initial implementation of `protocol/tx`: I briefly changed `entry` to a `struct` containing a hash and an `interface` rather than an `interface` itself, and you said you preferred `entry` to be the `interface`. We _could_ keep `entry` as an interface and just add a `Hash` method to it, implementing it identically for every entry type, but that seems strictly worse than having `EntryRef`.